### DISCUSSION TITLE
How can I display a list of categories without the content being displayed on the page like a blog?

### DISCUSSION THREAD

**Discussion by Julzn1014** [quarto-dev/quarto-cli#9967] [Q&A]
### Description

I am trying to build a example website for a project. 
I have programmed a table with Python and itables, which should be displayed on the start page. 
Now the categories should be displayed on the right side like in a blog, so that you can directly see which categories are available and click on them like in a blog. 
However, if I use the function 
listing: 
     categories: true 
the preview text of the pages is displayed below the table, just like on a blog. However, I don't need this content as it is distracting. Only the table should be displayed as content on the start page. 

Is there a way to display only the listing of the categories on the right without displaying the content under the table? 

This is the content of the index.qmd page:

```qmd
---
title: Data
subtitle: some Examples
lang: de

listing:
  categories: true

output:
  quarto::quarto_html:
    toc: true
    css: styles.css
   
execute:
  echo: false

html-table-processing: none 


---
```{python}



from itables import show
import pandas as pd 
import os

def find_word_in_files(folder_path):
   
    for root, dirs, files in os.walk(folder_path):
        data = []
        for file_name in files:
           
            file_path = os.path.join(root, file_name)
            try:
                
                with open(file_path, 'r') as file:
         
                    lines = file.readlines()
              
                    link = '<a href="./data/'+(file_name.split('.'))[0]+'.html">'+(lines[1].split())[1].strip('\"')+'</a>'

    
                    lineTwo = (lines[2].split())[1].strip('\"')

                    lineThree = (lines[3].split())[1]

                    data.append([link,lineTwo,lineThree])

                
            except Exception as e:
                continue
             
    
        return data


df = pd.DataFrame(find_word_in_files(folder_path='data'), columns=["Vertragspartner Bräutigam", "Vertragspartner Braut", "Datum"])  



show(df, lengthMenu=[10, 25, 50],language={'lengthMenu':'_MENU_ Einträge pro Seite', 'zeroRecords':'Keine Einträge gefunden','info':'Seite _PAGE_ von _PAGES_', 'infoEmpty':'Keine Einträge verfügbar','infoFiltered':'(gefiltert von _MAX_ Einträgen)', 'paginate':{'first':'Erste','last':'Letzte', 'next': 'Nächste','previous':'Vorherige'}, 'search':'Suche','loadingRecords':'Lade Einträge...', 'processing':'Verarbeite...'})  


```

If I want to use a custom listing for the categories, the page cannot be rendered if the table is created with Python. 

This is what the listing function looks like. The text is displayed below the table

<img width="433" alt="Bildschirmfoto 2024-06-12 um 11 16 54" src="https://github.com/quarto-dev/quarto-cli/assets/126703908/1bc1aadf-8f9e-48d1-9810-354b14e9b86f">

Thanks in advance for your help!

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs as it was described in the issue when you created it? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by Julzn1014** [reply]
Sorry, I forgot a ' for the formatting. 
But I don't see the need to split multiple files. The picture should only clarify where the problem lies.  If this is not wanted, it is best to leave it alone. 
If there is no answer to my problem of how to display only the categories without the content, it would be great to just say so.

**Reply by mcanouil** [reply]
You are assuming your issue can be solved from a screenshot which is most of the time is not true.
Under that assumption, I can provide the following dirty solution: use the CSS `display: none` on the whole content you don't want to display (use your favourite browser developer/inspect mode to know what is to be targeted).
Currently, I (and very likely others) cannot replicate your case using what you provided, thus it's hard to provide a real solution.

Providing properly formatted code helps other help you, and help others understand the issue so they can relate with their own issues.

For instance, the below part of your YAML is not Quarto syntax and is not documented anywhere in Quarto CLI documentation.

```
output:
  quarto::quarto_html:
    toc: true
    css: styles.css
```

This being said, the purpose of the listings is to display contents because that's a "listings of contents" which may or may not have categories that you can display.

Possible solutions:
- use custom listings
- use Lua filter

**Reply by Julzn1014** [reply]
No, I didn't think it could be solved with a screenshot. I was thinking more of clarifying where my problem lies. Namely, that text is still displayed there when I want to add the categories. 

And yes, you are right, the part of the code with "output" is not necessary for the execution and is just a remnant of a previous attempt I found in another blog, as there were no other hints for this problem. But it didn't lead to a solution either. 

We seem to have different approaches to how we explain problems. In the end, I just wanted to explain with these two parts where my problem is and if the error cannot be fixed this way, I was hoping for hints on what else is needed to get closer to the solution. 

I will try again with a custom listing and Lua filters and hope that there will be no more problems with rendering the table.

**Reply by mcanouil** [reply]
If you can manage to provide a small Git repositories which exhibit the issue you are having, it will help us figure out if something is going wrong, as intended, and/or possible workarounds.
